tests/unittests/test_backoff_error_handling.py

```diff
@@ -0,0 +1,1987 @@
+import unittest
+import socket
+from unittest import mock
+import tap_bing_ads
+from urllib.error import HTTPError, URLError
+import ssl
+from suds.transport import TransportError
+import datetime
+
+class MockClient():
+    '''Mocked ServiceClient class and it's method to pass the test case'''
+    def __init__(self, error):
+        self.error = error
+        self.count = 0
+
+    def GetCampaignsByAccountId(self, AccountId, CampaignType):
```

Comment:
Please provide a small comment for each function.

Reply:
Added comments.